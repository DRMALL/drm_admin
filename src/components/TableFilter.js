

import React from 'react'
import Select from 'antd/lib/select'

const TableFilter = ({ changeFun, arr }) => (
        <Select
          showSearch
          style={{ width: 200 }}
          placeholder="点击下拉列表选择或者输入搜索"
          Focus = { true }
          optionFilterProp="children"
          allowClear= { true }
          onChange={ changeFun }
          filterOption={(input, option) => option.props.children.toLowerCase().indexOf(input.toLowerCase()) >= 0}
        >
        {
          arr.map(item =>
            <Select.Option key={item._id} value={item._id} >{ item.name }</Select.Option>)
        }
        </Select>
  )

export default TableFilter
