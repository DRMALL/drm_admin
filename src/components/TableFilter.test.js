

import React from 'react'
import TableFilter from './TableFilter'
import { render } from 'enzyme'
import toJson from 'enzyme-to-json'

it('TableFilter test', () => {
  const fun = function(){}
  const arr = [{ _id:'dsae234', name: 'sdasr33' },{ _id:'dsae234', name: 'sdasr33' }]
  const wrapper = render(<TableFilter changeFun={ fun } arr={ arr } />)
  expect(toJson(wrapper)).toMatchSnapshot()
})
