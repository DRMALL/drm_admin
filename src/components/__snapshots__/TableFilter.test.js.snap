// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableFilter test 1`] = `
<div
  class="ant-select ant-select-enabled ant-select-allow-clear"
  style="width:200px;"
>
  <div
    aria-autocomplete="list"
    aria-expanded="false"
    aria-haspopup="true"
    class="ant-select-selection
              ant-select-selection--single"
    role="combobox"
    tabindex="0"
  >
    <div
      class="ant-select-selection__rendered"
    >
      <div
        class="ant-select-selection__placeholder"
        style="display:block;user-select:none;-webkit-user-select:none;"
        unselectable="unselectable"
      >
        筛选
      </div>
      <div
        class="ant-select-search ant-select-search--inline"
        style="display:none;"
      >
        <div
          class="ant-select-search__field__wrap"
        >
          <input
            class="ant-select-search__field"
            value=""
          />
          <span
            class="ant-select-search__field__mirror"
          >
             
          </span>
        </div>
      </div>
    </div>
    <span
      class="ant-select-selection__clear"
      style="user-select:none;-webkit-user-select:none;display:none;"
      unselectable="unselectable"
    />
    <span
      class="ant-select-arrow"
      style="user-select:none;-webkit-user-select:none;"
      unselectable="unselectable"
    >
      <b />
    </span>
  </div>
</div>
`;
